guard/src/utils/writer.rs

```diff
@@ -18,58 +19,90 @@ impl Default for Writer {
     }
 
-    pub fn err_to_stripped(self) -> Result<String, FromUtf8Error> {
+    pub fn err_to_stripped(self) -> crate::rules::Result<String> {
         match self.err {
-            WriteBuffer::Vec(vec) => String::from_utf8(strip_ansi_escapes::strip(vec).unwrap()),
+            WriteBuffer::Vec(vec) => match String::from_utf8(strip_ansi_escapes::strip(vec)?) {
+                Ok(s) => Ok(s),
+                Err(e) => Err(Error::InternalError(FromUtf8Error(e.utf8_error()))),
+            },
             WriteBuffer::File(mut file) => {
                 let mut data = String::new();
                 file.read_to_string(&mut data)
                     .expect("Unable to read from file");
 
-                String::from_utf8(strip_ansi_escapes::strip(data).unwrap())
+                match String::from_utf8(strip_ansi_escapes::strip(data)?) {
+                    Ok(s) => Ok(s),
+                    Err(e) => Err(Error::InternalError(FromUtf8Error(e.utf8_error()))),
+                }
             }
-            _ => unreachable!(),
+            WriteBuffer::Stdout(..) => Err(Error::InternalError(IncompatibleWriteError(
+                "Unable to call err_to_stripped() on a stdout buffer. Stdout is a write-only stream."
+                    .to_string(),
+            ))),
+            WriteBuffer::Stderr(..) => Err(Error::InternalError(IncompatibleWriteError(
+                "Unable to call err_to_stripped() on a stderr buffer. Stderr is a write-only stream."
+                    .to_string(),
+            ))),
         }
     }
 
-    pub fn into_string(self) -> Result<String, FromUtf8Error> {
+    pub fn into_string(self) -> crate::rules::Result<String> {
         self.buffer.into_string()
     }
 
-    pub fn stripped(self) -> Result<String, FromUtf8Error> {
+    pub fn stripped(self) -> crate::rules::Result<String> {
         match self.buffer {
-            WriteBuffer::Vec(vec) => String::from_utf8(strip_ansi_escapes::strip(vec).unwrap()),
+            WriteBuffer::Vec(vec) => {
+                let stripped = strip_ansi_escapes::strip(vec);
+                match String::from_utf8(stripped?) {
+                    Ok(s) => Ok(s),
+                    Err(e) => Err(Error::InternalError(FromUtf8Error(e.utf8_error()))),
+                }
+            }
             WriteBuffer::File(mut file) => {
                 let mut data = String::new();
                 file.read_to_string(&mut data)
                     .expect("Unable to read from file");
 
-                String::from_utf8(strip_ansi_escapes::strip(data).unwrap())
+                let stripped = strip_ansi_escapes::strip(data.into_bytes());
+                match String::from_utf8(stripped?) {
+                    Ok(s) => Ok(s),
+                    Err(e) => Err(Error::InternalError(FromUtf8Error(e.utf8_error()))),
+                }
             }
-            _ => unreachable!(),
+            WriteBuffer::Stdout(..) => Err(Error::InternalError(IncompatibleWriteError(
```

Comment:
similar comment to previous function about needing a new error to better represent this